roles/pulp_common/handlers/restart_all_pulp.yml

```diff
@@ -1,30 +1,34 @@
 ---
-- name: Set state of pulpcore app (block)
-  block:
-    - name: Restart pulpcore service
-      systemd:
-        daemon_reload: true
-        name: "pulpcore.service"
-        state: restarted
-      become: true
-  rescue:
-    - name: Restart pulpcore service (retry)
-      systemd:
-        daemon_reload: true
-        name: "pulpcore.service"
-        state: restarted
-      become: true
-      retries: 5
-      delay: 12
-      register: result_restart_pulpcore
-      until: result is succeeded
-      tags:
-        - molecule-notest
+# Note: blocks (for "rescue") do not work in handler included files
+- name: Restart pulpcore service
+  systemd:
+    daemon_reload: true
+    name: "pulpcore.service"
+    state: restarted
+  become: true
+  ignore_errors: true
+  register: result_restart_pulpcore
+
+- name: Restart pulpcore service (retry)
+  systemd:
+    daemon_reload: true
+    name: "pulpcore.service"
+    state: restarted
+  become: true
+  retries: 5
+  delay: 12
+  register: result_restarts_pulpcore
+  until: result_restarts_pulpcore is succeeded
+  when:
+    - result_restart_pulpcore is failed
+  tags:
+    - molecule-notest
```

Comment:
why?

Reply:
If it fails the 1st time in CI, we do not want retry. Instead, we want to "Print status of pulpcore services and fail". So as to debug it. The retries are for actual users.